serving/using-external-dns-on-gcp.md

```diff
@@ -0,0 +1,378 @@
+1. Configure Knative to use your custom domain.
+```shell
+kubectl edit cm config-domain --namespace knative-serving
+```
+This command opens your default text editor and allows you to edit the config 
+map.
+```
+apiVersion: v1
+data:
+  example.com: ""
+kind: ConfigMap
+[...]
+```
+Edit the file to replace `example.com` with your custom domain (the 
+value of `$CUSTOM_DOMAIN`) and save your changes. In this example, we use domain `external-dns-test.my-org.do`
+for all routes:
+```
+apiVersion: v1
+data:
+  external-dns-test.my-org.do: ""
+kind: ConfigMap
+[...]
+```
+
+## Set up ExternalDNS
+
+This guide uses Google Cloud Platform as an example to show how to set up 
+ExternalDNS. You can find detailed instructions for other cloud providers in the
+[ExternalDNS documentation](https://github.com/kubernetes-incubator/external-dns#deploying-to-a-cluster).
+
+### Create a DNS zone for managing DNS records
+
+Skip this step if you already have a zone for managing the DNS records of your 
+custom domain.
+
+A DNS zone which will contain the managed DNS records needs to be created.
+
+Use the following command to create a DNS zone with [Google Cloud DNS](https://cloud.google.com/dns/):
+```shell
+export DNS_ZONE_NAME=<dns-zone-name>
+
+gcloud dns managed-zones create $DNS_ZONE_NAME \
+    --dns-name $CUSTOM_DOMAIN \
+    --description "Automatically managed zone by kubernetes.io/external-dns"
+```
+Make a note of the nameservers that were assigned to your new zone.
+```shell
+gcloud dns record-sets list \
+    --zone $DNS_ZONE_NAME \
+    --name $CUSTOM_DOMAIN \
+    --type NS
+```
+You should see output similar to the following assuming your custom domain is 
+`external-dns-test.my-org.do`:
+```
+NAME                             TYPE  TTL    DATA
+external-dns-test.my-org.do.  NS    21600  ns-cloud-e1.googledomains.com.,ns-cloud-e2.googledomains.com.,ns-cloud-e3.googledomains.com.,ns-cloud-e4.googledomains.com.
+```
+In this case, the DNS nameservers are `ns-cloud-{e1-e4}.googledomains.com`. 
+Yours could differ slightly, e.g. {a1-a4}, {b1-b4} etc.
+
+If this zone has the parent zone, you need to add NS records of this zone into 
+the parent zone so that this zone can be found from the parent.
+Assuming the parent zone is `my-org-do` and the parent domain is `my-org.do`, 
+and the parent zone is also hosted at Google Cloud DNS, you can follow these 
+steps to add the NS records of this zone into the parent zone: 
+```shell
+gcloud dns record-sets transaction start --zone "my-org-do"
+gcloud dns record-sets transaction add ns-cloud-e{1..4}.googledomains.com. \
+    --name "external-dns-test.my-org.do." --ttl 300 --type NS --zone "my-org-do"
+gcloud dns record-sets transaction execute --zone "my-org-do"
+```
+
+### Deploy ExternalDNS
+
+Firstly, choose the manifest of ExternalDNS.
+
+Use below manifest if you set up your cluster with [CloudDNS scope](#cluster-with-cloud-dns-scope).
+```
```

Comment:
add yaml here for correct rendering and below.

Reply:
Done